opsdroid/connector/slack/connector.py

```diff
@@ -0,0 +1,388 @@
+"""A connector for Slack."""
+import json
+import logging
+import os
+import re
+import ssl
+
+import aiohttp
+import certifi
+import opsdroid.events
+from emoji import demojize
+from opsdroid.connector import Connector, register_event
+from opsdroid.connector.slack.create_events import SlackEventCreator
+from opsdroid.connector.slack.events import Blocks, EditedBlocks
+from slack_sdk.errors import SlackApiError
+from slack_sdk.socket_mode.aiohttp import SocketModeClient
+from slack_sdk.socket_mode.request import SocketModeRequest
+from slack_sdk.socket_mode.response import SocketModeResponse
+from slack_sdk.web.async_client import AsyncWebClient
+from voluptuous import Required
+
+_LOGGER = logging.getLogger(__name__)
+
+_USE_BOT_TOKEN_MSG = (
+    "Please upgrade your config with a new Slack 'bot-token' instead of the classic 'token'. "
+    "Startig v0.22.0 RTM support has been dropped in favour of Socket Mode. "
+    "Please check Slack Connector docs for instructions on how to migrate. "
+    "https://docs.opsdroid.dev/en/stable/connectors/slack.html"
+)
+CONFIG_SCHEMA = {
+    Required("bot-token", msg=_USE_BOT_TOKEN_MSG): str,
+    "socket-mode": bool,
+    "app-token": str,
+    "bot-name": str,
+    "default-room": str,
+    "icon-emoji": str,
+    "start_thread": bool,
+}
+
+
+class ConnectorSlack(Connector):
+    """A connector for Slack."""
+
+    def __init__(self, config, opsdroid=None):
+        """Create the connector."""
+        super().__init__(config, opsdroid=opsdroid)
+        _LOGGER.debug(_("Starting Slack connector."))
+        self.name = "slack"
+        self.bot_token = config["bot-token"]
+        self.bot_name = config.get("bot-name", "opsdroid")
+        self.default_target = config.get("default-room", "#general")
+        self.icon_emoji = config.get("icon-emoji", ":robot_face:")
+        self.start_thread = config.get("start_thread", False)
+        self.socket_mode = config.get("socket-mode", True)
+        self.app_token = config.get("app-token")
+        self.ssl_context = ssl.create_default_context(cafile=certifi.where())
+        self.slack_web_client = AsyncWebClient(
+            token=self.bot_token,
+            ssl=self.ssl_context,
+            proxy=os.environ.get("HTTPS_PROXY"),
+        )
+        self.socket_mode_client = (
+            SocketModeClient(self.app_token, web_client=self.slack_web_client)
+            if self.app_token
+            else None
+        )
+        self.auth_info = None
+        self.user_info = None
+        self.bot_id = None
+        self.known_users = {}
+
+        self._event_creator = SlackEventCreator(self)
+
+        _LOGGER.error(_("The Slack Connector will not be available."))
```

Comment:
@jacobtomlinson  the Error: ```ERROR opsdroid.connector.slack.connector: The Slack Connector will not be available.``` you are getting is because of this line. It's not needed, not quite sure how it slipped there but I will remove it.

Reply:
Fixed